index_node/solr.py

```diff
@@ -87,20 +84,20 @@ def start_solr(solr_config_type, port_number, SOLR_INSTALL_DIR="/usr/local/solr"
     else:
         enable_nodes = "'-Denable.master=true -Denable.slave=true'"
 
-    start_solr_command = "{SOLR_INSTALL_DIR}/bin/solr start -d {SOLR_INSTALL_DIR}/server -s {SOLR_HOME}/{solr_config_type}-{port_number} -p {port_number} -a {enable_nodes} -m 512m".format(SOLR_INSTALL_DIR=SOLR_INSTALL_DIR, SOLR_HOME=SOLR_HOME, solr_config_type=solr_config_type, port_number=port_number, enable_nodes=enable_nodes)
-    print "start solr command:", start_solr_command
-    esg_functions.stream_subprocess_output(start_solr_command)
+    server_directory = "{}/server".format(SOLR_INSTALL_DIR)
+    solr_solr_home = "{}/{}-{}".format(SOLR_HOME, solr_config_type, port_number)
+    start_solr_options = ["start", "-d", server_directory, "-s", solr_solr_home, "-p", port_number, "-a", enable_nodes, "-m", "512m"]
+    esg_functions.call_binary("/usr/local/bin/solr", start_solr_options)
 
     solr_status(SOLR_INSTALL_DIR)
 
 def solr_status(SOLR_INSTALL_DIR):
     '''Check the status of solr'''
     try:
-        esg_functions.stream_subprocess_output("{SOLR_INSTALL_DIR}/bin/solr status".format(SOLR_INSTALL_DIR=SOLR_INSTALL_DIR))
-    except OSError:
-        pass
-    except SubprocessError:
-        raise("Error checking solr status")
+        esg_functions.call_binary("/usr/local/bin/solr", ["status"])
+    except ProcessExecutionError:
```

Comment:
Is there a difference between `/usr/local/bin/solr` and the `/usr/local/solr/bin/solr` used elsewhere?

Reply:
/usr/local/solr/bin/solr  is the correct path.  The other is a typo.  I'll push a commit to fix

Reply:
Sounds good

Reply:
@nathanlcarlson this PR is ready to go